server/chat/background/rca_prompt_builder.py

```
@@ -296,272 +298,99 @@ def get_user_providers(user_id: str) -> List[str]:
-                    {
-                        "service": r[0], "environment": r[1], "result": r[2],
-                        "build_number": r[3], "build_url": r[4], "commit_sha": r[5] or "",
-                        "branch": r[6], "deployer": r[7], "trace_id": r[8],
-                        "webhook_received_at": r[9].isoformat() if r[9] else None,
-                    }
-                    for r in rows
-                ]
-    except Exception as e:
-        logger.warning(f"Error fetching recent Jenkins deployments: {e}")
-        return []
+    def _collect(obj: Any, depth: int = 0) -> None:
```

Ruff (0.15.14)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `obj`

(ANN401)